tests/Database/Mysql/Feature/MysqlConnectionDriverTest.php

```diff
@@ -47,6 +47,7 @@ protected function registerDatabaseListener()
 
         $this->configureBoth(function ($event) {
             $event->useConnection('mysql', $event->configuration);
+            $event->configuration = $event->defaults($event->tenant, $event->configuration);
```

Comment:
I'm not sure I like the pattern of passing the configuration in, only to assign back to it later.

I see a couple options.  Either:
1. call defaults first without the configuration, and have it simply return an array with the values set, then pass that in to `useConnection` to override
2. pass the configuration into `defaults` by-reference, and just modify in-place instead of returning it
3. make `$event->useConnection()` return something that you can compose with (i.e. `$event->useConnection(.....)->withDefaults()`)

Thoughts?

Reply:
Totally agree.
1) Seems decent and okay behaviour to me.
2) Would refrain from doing pass by reference for this function
3) Feels like kind of overkill to me in this specific case and in that case we'd have to change all functions